plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts

```diff
@@ -469,11 +482,15 @@ export function addHistoricalEventsExportCapabilityV2(
                     plugin: pluginConfig.plugin?.name ?? '?',
                 })
             } catch (error) {
+                clearInterval(interval)
+
                 await handleExportError(error, activeExportParameters, payload, events.length)
                 return
             }
         }
 
+        clearInterval(interval)
```

Comment:
non-blocking

I guess it's possible but not probable that there is a status update running at the point this is called, and therefore could end up running multiple concurrent updates to the set key.

I would like us to eventually handle concurrency (and guarding against it) a little more deliberately but I think ok for now.

Reply:
What would be the problem of concurrent updates to the set key? The db should be handling those gracefully, and as long as we have updated within a reasonable time, that should be enough to stop resumes from happening.

Reply:
It's possible e.g. that you have the updates arrive out of order, and therefore the chunk will be marked as not complete. There isn't anything in the db that can say these updates should have happened in a different order afaict. 

I think the likelihood of this happening here is pretty low, but nonetheless without much more effort we can ensure consistency in these types of cases. Not saying we do this here though, but for green field developments or if this does in fact become an issue I'd challenge ourselves to think about these cases.

Reply:
disclaimer: I haven't looked through the code here in much depth so it might be way off the mark 😅 . This is 100% non-blocking on the PR.

Reply:
I think it may cause issues if updates are significantly delayed as we could for example overwrite a `done: true` with `done: false`.

We could mitigate this a bit by only setting the `statusTime` instead of the entire payload. This is enough to stop resumes for happening and should also protect against concurrent updates as the job only looks at the cursor to figure out the state of the job, doesn't care about the status time.

Reply:
Ah, it would nice to do that but it's not possible with the current implementation: we are moving an entire payload around at a time as a single JSON blob in a single PG column. We can't update just 1 key.

Reply:
Actually we can do it relatively easily by adding a new `setJson` method to our storage and then running a query that uses `UPDATE ... SET <key> = json_set(...)`. Then the `setInterval` function can call that instead of `set`.

I'll keep this in mind in case we run into issues.